include/portfolio_solver.h

```diff
@@ -0,0 +1,85 @@
+/*********************                                                        */
+/*! \file portfolio_solver.h
+** \verbatim
+** Top contributors (to current version):
+**   Amalee Wilson
+** This file is part of the smt-switch project.
+** Copyright (c) 2021 by the authors listed in the file AUTHORS
+** in the top-level source directory) and their institutional affiliations.
+** All rights reserved.  See the file LICENSE in the top-level source
+** directory for licensing information.\endverbatim
+**
+** \brief Implementation of a portfolio solving function that takes a vector
+**        of solvers and a term, and returns check_sat from the first solver
+**        that finishes.
+**/
+#pragma once
+
+#include <condition_variable>
+#include <mutex>
+#include <thread>
+
+#include "smt.h"
+
+namespace smt {
+
+// Used for synchronization.
+std::mutex m;
```

Comment:
how about moving the implementation in a separate cpp file? Doing so, the global variables will moved to the cpp file .

Reply:
I'm not super opposed to separating it into a header/implementation file. My hesitation was that it's super tiny and not a class, so why split it?

Reply:
the problem is that when we include this header file, those global variables will be visible, which I think is not intended

Reply:
That's the best point